### PR TITLE
LPD-34347 Add new options to apply JS Client Extensions to Admin pages

### PR DESCRIPTION
**Pull request by thiago-buarqque**
Ticket: https://liferay.atlassian.net/browse/LPD-34347

## Motivation
JS Client Extensions can be applied only to the Pages scope. The goal of this development is to provide the ability to apply them also to the Admin scope (instance)

## Proposed Solution
Add two new properties to scope JS client extensions to the instance level.

1. `scope`: State in which scope the JS client extension will be applied to (for now, either `page` (default) or `instance`).
2. `restrictedToAdminPages`: State that the JS client extension is applied only to admin pages (i.e. control panel pages). This options is useless without scope set to instance.

## Steps to Verify
1. Go to liferay-portal/workspaces/liferay-sample-workspace/client-extensions/liferay-sample-global-js-1 and add these attributes:

```yaml
    scope: "instance"
    scriptLocation: "head"
    restrictedToAdminPages: true
```

2. Deploy it.
3. Go to instance settings > feature flags > beta and enable LPD-30371.
4. Reload the page.

## Screenshots/videos (if applies)
[Images/videos that showcase the changes you've made](https://github.com/user-attachments/assets/4cd9e1d3-79e5-4e06-90ef-a5e2eaf445c6)

## Alternative Solutions that were discarded
Product considered having a new instance configuration (see [previous PR](https://github.com/liferay-platform-experience/liferay-portal/pull/692)), but that turned out to be over-engineering and time-consuming considering how long we have to deliver this MVP.

### CODE REVIEW

**Comment by thiago-buarqque** [comment]
ci:test:sf

**Comment by thiago-buarqque** [comment]
ci:test:relevant

**Comment by thiago-buarqque** [comment]
ci:test:workspaces

**Comment by thiago-buarqque** [comment]
ci:test:gradle-plugins

**Comment by thiago-buarqque** [comment]
ci:test:sf

**Comment by thiago-buarqque** [comment]
ci:test:gradle-plugins

**Comment by thiago-buarqque** [comment]
ci:test:workspaces

**Comment by thiago-buarqque** [comment]
ci:test:relevant

File: modules/sdk/gradle-plugins-workspace/src/gradleTest/clientExtensionTypeGlobalJSShouldFail/with-invalid-restrictedToAdminPages/client-extension.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+with-invalid-restrictedToAdminPages:
```
Comment (drewbrokke) [inline modules/sdk/gradle-plugins-workspace/src/gradleTest/clientExtensionTypeGlobalJSShouldFail/with-invalid-restrictedToAdminPages/client-extension.yaml:1]: I see `with-invalid-script-location` (all kebab-case) in another file, but you're using camelCase here. I'd say make it all kebab-case for consistency.

File: modules/sdk/gradle-plugins-workspace/src/main/java/com/liferay/gradle/plugins/workspace/configurator/ClientExtensionProjectConfigurator.java [hunk 2]
```diff
@@ -1206,6 +1212,27 @@ private void _validateRequiredTypeSettingsKeys(
...
+			(typeSettingsValue instanceof Boolean)) {
+
+			return;
+		}
+
+		throw new GradleException(
+			String.format(
+				"Client extension %s must have a boolean value for the " +
+					"property %s",
+				clientExtension.id, StringUtil.quote(typeSettingsKey)));
```
Comment (drewbrokke) [inline modules/sdk/gradle-plugins-workspace/src/main/java/com/liferay/gradle/plugins/workspace/configurator/ClientExtensionProjectConfigurator.java:1233]: What is the purpose of the `required` parameter here? That should be covered by `_validateRequiredTypeSettingsKeys` already. This one's just asserting a type, right?

File: modules/sdk/gradle-plugins-workspace/src/gradleTest/clientExtensionTypeGlobalJS/build.gradle [hunk 3]
```diff
@@ -1,15 +1,26 @@
...
 	}
+}
+private File _getClientExtensionConfigJSONFile(String projectName) {
+	Project clientExtensionProject = project(":${projectName}")
+
+	File clientExtensionConfigJSONFile = clientExtensionProject.file("${clientExtensionProject.getProjectDir()}/build/liferay-client-extension-build/${projectName}.client-extension-config.json")
+
+	assert clientExtensionConfigJSONFile.exists()
+
+	return clientExtensionConfigJSONFile
```
Comment (drewbrokke) [inline modules/sdk/gradle-plugins-workspace/src/gradleTest/clientExtensionTypeGlobalJS/build.gradle:25]: Could this one be split up into a couple of cases that utilize `testClientExtensionBuildFileContentsTestCase.gradle` such as [testConfigJson.gradle](https://github.com/liferay/liferay-portal/blob/master/modules/sdk/gradle-plugins-workspace/src/gradleTest/clientExtensionWithVirtualInstanceId/testConfigJson.gradle?plain=1#L11)

**Review by drewbrokke — CHANGES_REQUESTED** [review]
Hey @thiago-buarqque thanks for sending this. A couple of comments, but overall it looks pretty straightforward.

File: modules/sdk/gradle-plugins-workspace/src/main/java/com/liferay/gradle/plugins/workspace/configurator/ClientExtensionProjectConfigurator.java (hunk 2 above)
Comment (thiago-buarqque) [reply modules/sdk/gradle-plugins-workspace/src/main/java/com/liferay/gradle/plugins/workspace/configurator/ClientExtensionProjectConfigurator.java:1233]: You are right, thanks for the heads up

**Comment by thiago-buarqque** [comment]
ci:test:sf

**Comment by thiago-buarqque** [comment]
ci:test:relevant

**Comment by thiago-buarqque** [comment]
ci:test:workspaces

**Comment by thiago-buarqque** [comment]
Changes made to a new commit. Please @drewbrokke, let me know if you have any other considerations

**Comment by thiago-buarqque** [comment]
ci:test:workspaces

**Comment by thiago-buarqque** [comment]
ci:test:relevant

**Comment by thiago-buarqque** [comment]
ci:test:gradle-plugins

**Comment by thiago-buarqque** [comment]
ci:stop

**Comment by thiago-buarqque** [comment]
ci:test:gradle-plugins

**Comment by thiago-buarqque** [comment]
ci:test:relevant

**Comment by thiago-buarqque** [comment]
@drewbrokke, I am not sure why `ci:test:gradle-plugins` is failing. It passes locally

**Comment by ethib137** [comment]
ci:report:78036561

File: portal-impl/src/portal.properties [hunk 4]
```diff
@@ -6972,6 +6977,11 @@
     #
     feature.flag.LPD-29516.type=beta
 
+    #
+    # Env: LIFERAY_FEATURE_PERIOD_FLAG_PERIOD__UPPERCASEL__UPPERCASEP__UPPERCASED__MINUS__NUMBER3__NUMBER0__NUMBER3__NUMBER7__NUMBER1__PERIOD_TYPE
+    #
+    feature.flag.LPD-30371.type=beta
```
Comment (ethib137) [inline portal-impl/src/portal.properties:6983]: Please make this a dev feature flag. We should only switch it to beta after product reviews the development in master.

**Review by ethib137 — CHANGES_REQUESTED** [review]
Nice work @thiago-buarqque!

**Comment by thiago-buarqque** [comment]
ci:test:sf

**Comment by thiago-buarqque** [comment]
ci:test:sf

**Review by drewbrokke — APPROVED** [review]

**Comment by thiago-buarqque** [comment]
ci:forward:force

**Comment by thiago-buarqque** [comment]
ci:report:79318850

**Comment by thiago-buarqque** [comment]
ci:test:stable

**Comment by thiago-buarqque** [comment]
ci:forward:force

**Pull request closed without merging**